src/core/storage/fileio/fs_utils.cpp

```diff
@@ -440,6 +440,8 @@ EXPORT std::string get_protocol(std::string path) {
   size_t proto = path.find("://");
   if (proto != std::string::npos) {
     std::string ret = boost::algorithm::to_lower_copy(path.substr(0, proto));
+    std::transform(ret.begin(), ret.end(), ret.begin(),
+                   [](unsigned char c) { return std::tolower(c); });
```

Comment:
Doesn't the to_lower_copy do the lowercase already?

Reply:
Ah, let me check. I have un-pushed changes. Let's find the branch the resume my work.

Reply:
this does everything inline, without an extra copy.

Reply:
Ah, but I'm referring to the `boost::algorithm::to_lower_copy(path.substr(0, proto));` call above.

Reply:
https://www.boost.org/doc/libs/1_72_0/doc/html/boost/algorithm/to_lower_copy.html

Reply:
I see! The name is quite confusing! I will switch to to_lower_copy.